src/pages/ReimbursementAccount/ValidationStep.js

```diff
@@ -47,82 +45,29 @@ class ValidationStep extends React.Component {
         super(props);
 
         this.submit = this.submit.bind(this);
-
-        this.state = {
-            amount1: ReimbursementAccountUtils.getDefaultStateForField(props, 'amount1', ''),
-            amount2: ReimbursementAccountUtils.getDefaultStateForField(props, 'amount2', ''),
-            amount3: ReimbursementAccountUtils.getDefaultStateForField(props, 'amount3', ''),
-        };
-
-        this.requiredFields = [
-            'amount1',
-            'amount2',
-            'amount3',
-        ];
-
-        this.errorTranslationKeys = {
-            amount1: 'common.error.invalidAmount',
-            amount2: 'common.error.invalidAmount',
-            amount3: 'common.error.invalidAmount',
-        };
-
-        this.getErrors = () => ReimbursementAccountUtils.getErrors(this.props);
-        this.getErrorText = inputKey => ReimbursementAccountUtils.getErrorText(this.props, this.errorTranslationKeys, inputKey);
-        this.clearError = inputKey => ReimbursementAccountUtils.clearError(this.props, inputKey);
-    }
-
-    /**
-    * @param {Object} value
-    */
-    setValue(value) {
-        BankAccounts.updateReimbursementAccountDraft(value);
-        this.setState(value);
-    }
-
-    /**
-     * Clear the error associated to inputKey if found and store the inputKey new value in the state.
-     *
-     * @param {String} inputKey
-     * @param {String} value
-     */
-    clearErrorAndSetValue(inputKey, value) {
-        this.setValue({[inputKey]: value});
-        this.clearError(inputKey);
+        this.validate = this.validate.bind(this);
     }
 
     /**
+     * @param {Object} values - form input values passed by the Form component
      * @returns {Boolean}
      */
-    validate() {
+    validate(values) {
         const errors = {};
-        const values = {
-            amount1: this.filterInput(this.state.amount1),
-            amount2: this.filterInput(this.state.amount2),
-            amount3: this.filterInput(this.state.amount3),
-        };
 
-        _.each(this.requiredFields, (inputKey) => {
-            if (ValidationUtils.isRequiredFulfilled(values[inputKey])) {
+        _.chain(values).keys().each((inputKey) => {
+            const filteredValue = this.filterInput(values[inputKey]);
+            if (ValidationUtils.isRequiredFulfilled(filteredValue)) {
```

Comment:
NAB: I think alternatively we can use `_.each` to iterate the pairs of key-value of an object
```suggestion
        _.each(values, (value, inputKey) => {
            if (ValidationUtils.isRequiredFulfilled(this.filterInput(value))) {
```